File: front_end/lib/src/models/contact_dto.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:front_end/src/models/address_model.dart';

@JsonSerializable()
class ContactDto {
  ContactDto(
      {required this.addresses, required this.phones, required this.emails});

  List<AddressDto> addresses;
  List<String> phones;
  List<String> emails;

  factory ContactDto.fromJson(Map<String, dynamic> jsonString) {
    List<dynamic> phones = jsonString['phones'] ?? [];
    List<dynamic> emails = jsonString['emails'] ?? [];
    var inner = jsonString['addresses'] as List<dynamic>;
    List<AddressDto> addresses =
        inner.map((i) => AddressDto.fromJson(i)).toList();

    return ContactDto(
      addresses: addresses,
      phones: List<String>.from(phones.map((dynamic item) => item.toString())),
      emails: List<String>.from(emails.map((dynamic item) => item.toString())),
    );
  }

  Map<String, dynamic> toJson() =>
      {'addresses': addresses, 'phones': phones, 'emails': emails};
}

File: gambit/src/models/Response/Contact/ContactDto.ts
import AddressDto from "./AddressDto";

class ContactDto {
  addresses: Array<AddressDto> ;
  emails: Array<string>;
  phones: Array<string> ;

  constructor(addresses: Array<AddressDto>, email: Array<string>, phones: Array<string>)
  {
    this.addresses = addresses;
    this.emails = email;
    this.phones = phones;
  }
}

export default ContactDto;